### PR TITLE
[Feature][API] add common options

### PR DESCRIPTION
**Pull request by EricJoy2048**
sub tasks of https://github.com/apache/incubator-seatunnel/issues/3337

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactory.java [hunk 1]
```diff
@@ -32,11 +33,28 @@
...
+        OptionRule optionRule = optionRule();
+        if (optionRule == null) {
+            throw new FactoryException("OptionRule can not be null");
+        }
+
+        OptionRule sinkCommonOptionRule =
+            OptionRule.builder().optional(SinkCommonOptions.SOURCE_TABLE_NAME, SinkCommonOptions.PARALLELISM).build();
+        optionRule.getOptionalOptions().addAll(sinkCommonOptionRule.getOptionalOptions());
+        return optionRule;
+    }
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactory.java]: IMO, move this code snippet to the Util class

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSourceFactory.java [hunk 2]
```diff
@@ -36,4 +37,21 @@ public interface TableSourceFactory extends Factory {
...
+        OptionRule optionRule = optionRule();
+        if (optionRule == null) {
+            throw new FactoryException("OptionRule can not be null");
+        }
+
+        OptionRule sourceCommonOptionRule =
+            OptionRule.builder().optional(SourceCommonOptions.RESULT_TABLE_NAME, SourceCommonOptions.PARALLELISM).build();
+        optionRule.getOptionalOptions().addAll(sourceCommonOptionRule.getOptionalOptions());
+        return optionRule;
+    }
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSourceFactory.java]: Same as `TableSinkFactory`

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactory.java (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactory.java]: > IMO, move this code snippet to the Util class

Ok, I will do it after we removed V1 code from branch dev.

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServerStarter.java [hunk 3]
```diff
@@ -39,6 +40,13 @@ public static HazelcastInstanceImpl createHazelcastInstance(String clusterName)
...
+    public static HazelcastInstanceImpl createHazelcastInstance(@NonNull SeaTunnelConfig seaTunnelConfig) {
+        return ((HazelcastInstanceProxy) HazelcastInstanceFactory.newHazelcastInstance(
+            seaTunnelConfig.getHazelcastConfig(),
+            HazelcastInstanceFactory.createInstanceName(seaTunnelConfig.getHazelcastConfig()),
+            new SeaTunnelNodeContext(seaTunnelConfig))).getOriginal();
+    }
+
     public static HazelcastInstanceImpl createHazelcastInstance() {
         SeaTunnelConfig seaTunnelConfig = ConfigProvider.locateAndGetSeaTunnelConfig();
         return ((HazelcastInstanceProxy) HazelcastInstanceFactory.newHazelcastInstance(
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServerStarter.java]: Remove deplicate code

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServerStarter.java]: done

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSourceFactory.java (hunk 2 above)
Comment (CalvinKirs) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSourceFactory.java]: +1

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactory.java (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactory.java]: done

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSourceFactory.java (hunk 2 above)
Comment (EricJoy2048) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSourceFactory.java]: done

**Review by CalvinKirs — DISMISSED** [review]

**Review by CalvinKirs — DISMISSED** [review]

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/FactoryUtil.java [hunk 4]
```diff
@@ -155,4 +159,36 @@ private static List<Factory> discoverFactories(ClassLoader classLoader) {
             throw new FactoryException("Could not load service provider for factories.", e);
         }
     }
+
+    /**
+     * This method is called by SeaTunnel Web to get the full option rule of a source.
+     * Please don't overwrite this method.
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/FactoryUtil.java]:
```suggestion
```

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/FactoryUtil.java [hunk 5]
```diff
@@ -155,4 +159,36 @@ private static List<Factory> discoverFactories(ClassLoader classLoader) {
...
+
+        OptionRule sourceCommonOptionRule =
+            OptionRule.builder().optional(SourceCommonOptions.RESULT_TABLE_NAME, SourceCommonOptions.PARALLELISM).build();
+        sourceOptionRule.getOptionalOptions().addAll(sourceCommonOptionRule.getOptionalOptions());
+        return sourceOptionRule;
+    }
+
+    /**
+     * This method is called by SeaTunnel Web to get the full option rule of a sink.
+     * Please don't overwrite this method.
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/FactoryUtil.java]:
```suggestion
```

**Review by Hisoka-X — APPROVED** [review]

**Review by ic4y — APPROVED** [review]

**Pull request merged**